apis/clusters/v1beta1/kafka_webhook.go

```diff
@@ -81,6 +82,37 @@ func (kv *kafkaValidator) ValidateCreate(ctx context.Context, obj runtime.Object
 		return err
 	}
 
+	if k.Spec.OnPremisesSpec != nil {
+		osDiskSizeMatched, err := regexp.Match(models.StorageRegExp, []byte(k.Spec.OnPremisesSpec.OSDiskSize))
+		if !osDiskSizeMatched || err != nil {
+			return fmt.Errorf("disk size field for node OS must fit pattern: %s",
+				models.StorageRegExp)
+		}
+
+		dataDiskSizeMatched, err := regexp.Match(models.StorageRegExp, []byte(k.Spec.OnPremisesSpec.DataDiskSize))
+		if !dataDiskSizeMatched || err != nil {
+			return fmt.Errorf("disk size field for storring cluster data must fit pattern: %s",
+				models.StorageRegExp)
+		}
+
+		nodeMemoryMatched, err := regexp.Match(models.MemoryRegExp, []byte(k.Spec.OnPremisesSpec.DataDiskSize))
+		if !nodeMemoryMatched || err != nil {
+			return fmt.Errorf("node memory field must fit pattern: %s",
+				models.MemoryRegExp)
+		}
+
+		if k.Spec.PrivateNetworkCluster {
+			if k.Spec.OnPremisesSpec.SSHGatewayCPU == 0 || k.Spec.OnPremisesSpec.SSHGatewayMemory == "" {
+				return fmt.Errorf("fields SSHGatewayCPU and SSHGatewayMemory must not be empty")
+			}
+			sshGatewayMemoryMatched, err := regexp.Match(models.MemoryRegExp, []byte(k.Spec.OnPremisesSpec.DataDiskSize))
+			if !sshGatewayMemoryMatched || err != nil {
+				return fmt.Errorf("ssh gateway memory field must fit pattern: %s",
+					models.MemoryRegExp)
+			}
+		}
```

Comment:
Can you move it to separate function to not duplicate such code a lot of times?

Reply:
Yep, will fix